lib/form/paragraph3.flow

```diff
@@ -515,72 +593,35 @@ renderParagraphEx(
-						addParaWordWidth(acc, getParaWordWidth(pw, WordPositionInLineMiddle()))
-					)
-				);
-				ignoreLetterspacing = containsStruct(s, IgnoreLetterspacingOnReflow());
-				infos = getParaLineInfos(
-					alignment,
-					wordGroups,
-					w,
-					indent,
-					isParagraphMarked,
-					rtl,
-					fitLongWords,
-					extractStruct(s, defaultFontSizeStyle).size,
-					ignoreLetterspacing,
-					interlineSpacing
-				);
-
-				nextDistinct(linesCountB, length(infos));
-				// We have to rerender even when lines are identical, at minimum to move things because of potential size changes
-				maxLineWidth = fold(infos, 0.0, \acc, info -> max(info.lineWidth, acc));
-				if (maxLineWidth > w) {
-					traceApi.trace(\-> "DONT_FIT: " + toString(maxLineWidth) + " > " + toString(w));
-				}
-
-				traceApi.traceAndNest(\-> "updateFn [" + id + "] lines SET START w: " + toString(w) + ". maxLineWidth " + toString(oldMaxLineWidth) + " -> " + toString(maxLineWidth) + ". lines " + toString(oldLines) + " -> " + toString(getValue(linesCountB)));
-				next(linesB, ParaLinesInfos(infos, maxLineWidth));
-				traceApi.unnestAndTrace(\-> "updateFn [" + id + "] lines SET END  w: " + toString(w) + ". maxLineWidth " + toString(oldMaxLineWidth) + " -> " + toString(maxLineWidth) + ". lines " + toString(oldLines) + " -> " + toString(getValue(linesCountB)));
-
-				traceApi.unnestAndTrace(\-> "updateFn [" + id + "] END   w: " + toString(w) + ". maxLineWidth " + toString(oldMaxLineWidth) + " -> " + toString(maxLineWidth) + ". lines " + toString(oldLines) + " -> " + toString(getValue(linesCountB)));
-			};
-
 			uns = concatA([
 				[
-					subscribe(awB, \v -> {
-						traceApi.trace(\-> "awB changed. calling updateFn. awB = " + toString(v));
-						updateFn(v);
-					}),
+					subscribe(awB, deferredUpdate),
 				],
 				if (renderInfo.inline || ndynamic == 0 || ndynamic == ndymamicUpper) [] else
 				[
 					subscribe2(update, \u -> {
 						// When an update from a dynamic element is requested, we do it in the next frame
-						if (u == 1) deferUntilNextFrameRendered(\ -> {
-							traceApi.trace(\-> "updateFn DEFERRED STARTED");
-							next(update, 0);
-							updateFn(getValue(awB))
-						})
+						if (u == 1) {
+							id = traceApi.makeRenderingId();
+							traceApi.trace(\-> "updateFn["+id+"] TRIGGERED");
+
+							deferUntilNextFrameRendered(\ -> {
+								traceApi.traceAndNest(\-> "updateFn["+id+"] DEFERRED START");
+								next(update, 0);
+								updateFn(id, getValue(awB));
+								traceApi.unnestAndTrace(\-> "updateFn["+id+"] DEFERRED END");
+							})
+						}
 					})
 				],
 				mapi(dynamicWords, \i, w -> makeSubscribe2(w.metrics, \m -> {
-					if (renderInfo.inline || i < ndymamicUpper) {
-						traceApi.trace(\-> "dynamicWordsMetrics IMMEDIATE updateFn for m = " + toString(m));
-						updateFn(getValue(awB))
+					id = traceApi.makeRenderingId();
+					if (true || renderInfo.inline || i < ndymamicUpper) {
```

Comment:
Is this true here by design or for debug reasons?

Reply:
debug, thank's. fixed